tests/testthat/test-test_dir.R

```diff
@@ -45,3 +45,23 @@ test_that('test_dir() helpers', {
   df <- as.data.frame(res)
   expect_true(all(!df$error & df$failed == 0))
 })
+
+test_that('find_test_scripts() with tricky names', {
```

Comment:
I'm no sure I understand.  I mocked `dir` so that I did not have to create a directory structure with the problematic file names.  Do you want me to create a new function that accepts the output of `dir` and does the filtering, and then modify `find_test_scripts` to use that function instead?